test/compilable/typeid_name.d

```diff
@@ -0,0 +1,14 @@
+
```

Comment:
A `fail_compilation` test should be added to cover the introduced error message.

Reply:
I'd rather not introduce an test for an error message which might have to be removed.
the restriction to `typeid().name` is not intentional. and if I'd have to remove a test when extending it, that would look bad.

Reply:
If the message is removed, then the test can be changed.

Reply:
We need coverage of all the lines of code introduced.